pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -598,6 +636,126 @@ func (r *defaultKialiReconciler) reconcileKiali(ctx context.Context, kialiCRName
 	return nil
 }
 
+type PrometheusReconciler interface {
+	reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, configuredMembers []string) error
+}
+
+type defaultPrometheusReconciler struct {
+	Client client.Client
+}
+
+func (r *defaultPrometheusReconciler) reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, members []string) error {
```

Comment:
It's quite a big function, so I would prefer to move it to another file in the same package and split it into smaller functions to improve readability. WDYT?